frontends/web/src/routes/account/walletconnect/dashboard.tsx

```diff
@@ -0,0 +1,113 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLoad } from '../../../hooks/api';
+import { SessionTypes } from '@walletconnect/types';
+import { getSdkError } from '@walletconnect/utils';
+import { route } from '../../../utils/route';
+import useInitialization, { getAddressFromEIPString, truncateAddress, web3wallet, } from './utils';
+import { IAccount, getReceiveAddressList } from '../../../api/account';
+import { Header, Main } from '../../../components/layout';
+import { View, ViewContent } from '../../../components/view/view';
+import { WCSessionCard } from './components/session-card/session-card';
+import { Button } from '../../../components/forms';
+import styles from './dashboard.module.css';
+
+type TProps = {
+  accounts: IAccount[];
+  code: string;
+}
+
+export const DashboardWalletConnect = ({ code, accounts }: TProps) => {
+  const isInitialized = useInitialization();
+  const { t } = useTranslation();
+  const [sessions, setSessions] = useState<SessionTypes.Struct[]>();
+  const receiveAddresses = useLoad(getReceiveAddressList(code));
+
+  useEffect(() => {
+    if (isInitialized) {
+      updateSessions();
+    }
+  }, [isInitialized]);
+
+  useEffect(() => {
+    if (isInitialized) {
+      web3wallet?.on('session_delete', updateSessions);
+      return () => {
+        web3wallet?.off('session_delete', updateSessions);
+      };
+    }
+  }, [isInitialized]);
+
+  const updateSessions = () => {
+    setSessions(Object.values(web3wallet?.getActiveSessions()));
+  };
+
+  const handleDisconnectSession = async (topic: string) => {
+    await web3wallet.disconnectSession({
+      topic,
+      reason: getSdkError('USER_DISCONNECTED'),
+    });
+    updateSessions();
+  };
+
+  if (!receiveAddresses || !isInitialized) {
+    return null;
+  }
+
+  const receiveAddress = truncateAddress(receiveAddresses[0].addresses[0].address);
+  const accountName = (accounts && accounts.find(acct => acct.code === code))?.name || '';
+  const hasSession = sessions && sessions.length > 0;
+
+  return (
+    <Main>
+      <Header
+        title={<h2>{t('walletConnect.walletConnect')}</h2>}
+      />
+      <View>
+        <ViewContent>
+          <div className={styles.headerContainer}>
+            <div>
+              <p>{accountName}</p>
+              <p>{receiveAddress}</p>
+            </div>
+            <Button onClick={() => route(`/account/${code}/wallet-connect/connect`)} primary>{t('walletConnect.dashboard.newConnection')}</Button>
+          </div>
+          <hr className={styles.separator} />
+          {hasSession &&
+            <div className={styles.sessionCardsContainer}>
+              <p>{t('walletConnect.dashboard.allSessions')}</p>
+              {sessions.map(s =>
+                <WCSessionCard
+                  key={s.topic}
```

Comment:
could you change `s` to `session`, and is `session.topic` unique?

Reply:
sure! and yes topic is unique.